File: genealogy-tree-API/GenealogyTree.Domain/DTO/SyncRequest/SyncRequestCreateUpdateModel.cs
using GenealogyTree.Domain.DTO.Person;
using GenealogyTree.Domain.DTO.User;

namespace GenealogyTree.Domain.DTO.SyncRequest
{
    public class SyncRequestCreateUpdateModel
    {
        public int? Id { get; set; }
        public int SenderUserId { get; set; }
        public UserDetailsModel ReceiverUser { get; set; }
        public PersonDetailsModel ReceiverReferenceInSenderTree { get; set; }
    }
}

File: gene-tree/src/app/modules/sync/models/sync-request-create-update.model.ts
import { PersonDetailsModel } from './../../person/models/person/person-details.model';
import { UserDetailsModel } from './../../user/models/user-details.model';
import { Guid } from 'guid-typescript';

export class SyncRequestCreateUpdateModel {
  id?: number;
  sunderUserId: Guid;
  receiverUser: UserDetailsModel;
  receiverReferenceInSenderTree: PersonDetailsModel;
}
